### PR TITLE
[FEAT] 1606 - Adding hour expression in date util

### PR DESCRIPTION
**Pull request by suriya-ganesh**
Adds `dt_hour()` Expression to fetch hour of the current day in a column. Based on, #1606

### CODE REVIEW

File: src/daft-core/src/array/ops/date.rs [hunk 1]
```diff
@@ -107,4 +107,40 @@ impl TimestampArray {
...
+                    })
+                },
+            ))),
+        }?;
+
+        Ok(UInt32Array::new(
+            std::sync::Arc::new(Field::new(self.name(), DataType::UInt32)),
+            Box::new(date_arrow),
+        )
+        .unwrap_or(UInt32Array::from(("hour", vec![0]))))
```
Comment (suriya-ganesh) [inline src/daft-core/src/array/ops/date.rs:144]: Maybe, I don't need the unwrap and just a `?` is enough?

Comment (samster25) [inline src/daft-core/src/array/ops/date.rs:144]: You shouldn't have to unwrap the `UInt32Array::new` and then wrap it in `Ok`. You should be able to return the result from UInt32Array::new directly! 
```
UInt32Array::new(
            std::sync::Arc::new(Field::new(self.name(), DataType::UInt32)),
            Box::new(date_arrow),
        )
```

Comment (suriya-ganesh) [reply src/daft-core/src/array/ops/date.rs]: Yea, not sure what I was thinking. Fixing it now.

**Comment by samster25** [comment]
looks great!

**Pull request merged**